ledger/sandbox/src/main/resources/db/migration/postgres/V16__Create_command_completions_table.sql

```diff
@@ -0,0 +1,18 @@
+-- Copyright (c) 2019 The DAML Authors. All rights reserved.
+-- SPDX-License-Identifier: Apache-2.0
+
+CREATE TABLE participant_command_completions
+(
+    completion_offset bigint not null,
+    record_time timestamp not null, -- TODO re-evaluate usefulness after new ledger time lands
+
+    application_id varchar, -- null for checkpoints
+    submitting_party varchar, -- null for checkpoints
+    command_id varchar, -- null for checkpoints
+
+    transaction_id varchar, -- null if the command was rejected and checkpoints
+    status_code integer, -- null for successful command and checkpoints
+    status_message varchar -- null for successful command and checkpoints
+);
+
+CREATE INDEX ON participant_command_completions(completion_offset, application_id, submitting_party);
```

Comment:
Perhaps the trailing newline would be nice.

Reply:
Yes, my formatter.

Reply:
Addressed by https://github.com/digital-asset/daml/pull/4681/commits/d9b20c8dcfa57f45549dfd1a61bd28d9cfe18b43.